File: app/src/main/java/org/worshipsongs/domain/Setting.java
package org.worshipsongs.domain;

import android.view.Display;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.apache.commons.lang3.builder.ToStringBuilder;

import java.util.Set;

/**
 * author: madasamy
 * version: 2.1.0
 */
public class Setting
{
    private static Setting setting;
    private int position;
    private int slidePosition;
    private Song song;

   
    private Setting()
    {

    }

    public static Setting getInstance()
    {
        if (setting == null) {
            setting = new Setting();
        }
        return setting;
    }

    public int getPosition()
    {
        return position;
    }

    public void setPosition(int position)
    {
        this.position = position;
    }

    public int getSlidePosition()
    {
        return slidePosition;
    }

    public void setSlidePosition(int slidePosition)
    {
        this.slidePosition = slidePosition;
    }

    public Song getSong()
    {
        return song;
    }

    public void setSong(Song song)
    {
        this.song = song;
    }

    @Override
    public String toString()
    {
        ToStringBuilder stringBuilder = new ToStringBuilder(this);
        stringBuilder.append("position", position);
        return stringBuilder.toString();
    }

    @Override
    public boolean equals(Object object)
    {
        if (object instanceof Setting) {
            Setting otherObject = (Setting) object;
            EqualsBuilder equalsBuilder = new EqualsBuilder();
            equalsBuilder.append(getPosition(), otherObject.getPosition());
            return equalsBuilder.isEquals();
        }
        return false;
    }

    @Override
    public int hashCode()
    {
        HashCodeBuilder hashCodeBuilder = new HashCodeBuilder();
        hashCodeBuilder.append(getPosition());
        return hashCodeBuilder.hashCode();
    }
}

File: app/src/main/java/org/worshipsongs/domain/Setting.kt
package org.worshipsongs.domain

import android.view.Display

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.apache.commons.lang3.builder.ToStringBuilder

/**
 * author: madasamy
 * version: 2.1.0
 */
class Setting private constructor()
{
    var position: Int = 0
    var slidePosition: Int = 0
    var song: Song? = null

    override fun toString(): String
    {
        val stringBuilder = ToStringBuilder(this)
        stringBuilder.append("position", position)
        return stringBuilder.toString()
    }

    override fun equals(`object`: Any?): Boolean
    {
        if (`object` is Setting)
        {
            val otherObject = `object` as Setting?
            val equalsBuilder = EqualsBuilder()
            equalsBuilder.append(position, otherObject!!.position)
            return equalsBuilder.isEquals
        }
        return false
    }

    override fun hashCode(): Int
    {
        val hashCodeBuilder = HashCodeBuilder()
        hashCodeBuilder.append(position)
        return hashCodeBuilder.hashCode()
    }

    companion object
    {
        val instance = Setting()
    }
}
